optimistix/_solver/optax.py

```diff
@@ -97,7 +97,14 @@ def step(
                 ("loss" in self.verbose, "Loss", f),
                 ("y" in self.verbose, "y", y),
             )
-        updates, new_opt_state = self.optim.update(grads, state.opt_state, y)
+
+        # fix args and discard aux
+        _fn_for_optax = lambda y: fn(y, args)[0]
+        _fn_for_optax = eqx.filter_closure_convert(_fn_for_optax, y)
```

Comment:
I'm surprised the closure-conversion is necessary. Optax itself doesn't return optimizers that are pytrees, is it really expecting that `value_fn` be both a callable and a pytree?

Reply:
You're right, it's not needed. I just removed it.